### ISSUE TITLE
export yolo 11 in ONXX with options doesn't change the accuracy and confidence score. are these option working in yolov11?

### ISSUE THREAD

**Issue by GISScience** [ultralytics/ultralytics#16549]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

export yolo 11 in ONXX with options doesn't change the accuracy at all?

i have tried onxx export with following commands -

**1.model = YOLO("/content/drive/MyDrive/YoloQuant/models/yolo11n-obb.pt")

2.model.export(format="onnx" , simplify=True, imgsz=(1024,1024), dynamic=True)

3.model.export(format="onnx" , half=True, imgsz=(1024,1024), batch=1)**

Export and predict using 1 , 2 , 3 is successfully and working properly but there is no accuracy drop and change in confidence score?

Are all these option working ?? becuase usually simplify=True,and half=True,  will change the accuracy and confidence score.

### Environment

Ultralytics 8.3.0 🚀 Python-3.10.12 torch-1.11.0+cu113 CUDA:0 (Tesla T4, 15102MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 46.2/112.6 GB disk)

### Minimal Reproducible Example

from ultralytics import YOLO
# OBB https://docs.ultralytics.com/tasks/obb/#val
# Load a model
model = YOLO("/content/drive/MyDrive/models/yolov8n-obb.pt")  # load a custom model

# Predict with the model
results = model("/content/drive/MyDrive/input.jpg",imgsz=(1024,1024))  # predict on an image
#results = model("https://ultralytics.com/images/bus.jpg")  # predict on an image
# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="/content/drive/MyDrive/output.jpg")  # save to disk

# Predict with the model using single  image
from ultralytics import YOLO
# Load the exported TensorRT model
tensorrt_model = YOLO("/content/drive/MyDrive/models/yolo11n-obb.onnx")
#tensorrt_model.info()
# Run inference
#results = tensorrt_model("https://ultralytics.com/images/bus.jpg")
results = tensorrt_model("/content/drive/MyDrive/inpu.jpg",imgsz=(1024,1024))  # predict on an image
#results = model("https://ultralytics.com/images/bus.jpg")  # predict on an image
# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="/content/drive/MyDrive/output.jpg")

### Additional

from ultralytics import YOLO
# OBB https://docs.ultralytics.com/tasks/obb/#val
# Load a model
model = YOLO("/content/drive/MyDrive/models/yolov8n-obb.pt")  # load a custom model

# Predict with the model
results = model("/content/drive/MyDrive/input.jpg",imgsz=(1024,1024))  # predict on an image
#results = model("https://ultralytics.com/images/bus.jpg")  # predict on an image
# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="/content/drive/MyDrive/output.jpg")  # save to disk

# Predict with the model using single  image
from ultralytics import YOLO
# Load the exported TensorRT model
tensorrt_model = YOLO("/content/drive/MyDrive/models/yolo11n-obb.onnx")
#tensorrt_model.info()
# Run inference
#results = tensorrt_model("https://ultralytics.com/images/bus.jpg")
results = tensorrt_model("/content/drive/MyDrive/inpu.jpg",imgsz=(1024,1024))  # predict on an image
#results = model("https://ultralytics.com/images/bus.jpg")  # predict on an image
# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="/content/drive/MyDrive/output.jpg")

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GISScience, thank you for reaching out and your interest in Ultralytics 🚀! This is an automated response, but don't worry—an Ultralytics engineer will assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com) if you're new, where you can explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. This might help you resolve common issues.

If this is a 🐛 Bug Report, please ensure that you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist in debugging. Your detailed example is already beneficial, thank you!

If this is a custom training ❓ Question, kindly provide relevant information such as dataset samples and training logs. Ensure you are adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Feel free to join the Ultralytics community where you feel most at home. For real-time interaction, check out our [Discord](https://ultralytics.com/discord) 🎧. For thorough discussions, visit [Discourse](https://community.ultralytics.com), or join discussions on our [Subreddit](https://reddit.com/r/ultralytics) to engage with diverse topics.

## Upgrade

To verify if the issue persists in the latest version, upgrade to the newest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) using [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) in your environment:

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 functions efficiently in these updated and validated environments, all equipped with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with GPU at no cost: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green status on this badge indicates successful [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests across all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu, checked daily and on each commit.

Thank you for your patience and contribution. Your willingness to submit a PR is greatly appreciated! 🙌

**Comment by Y-T-G** [comment]
If the ONNX model was created successfully, and the model loads and runs, then they're working as expected. Accuracy drop may or may not occur.

**Comment by glenn-jocher** [comment]
@GISScience also `simplify=True` is now a default, so explicitly defining it there won't have any effect. See https://docs.ultralytics.com/modes/export/#arguments

**Comment by GISScience** [comment]
@glenn-jocher @Y-T-G  During quantization of YOLOv11, the confidence score for the same object should demonstrably decrease slightly. However, I did not observe this drop. In both Method 1 (using half=True for quantization) and Method 2 (TensorFlow engine with int8=True), the confidence scores for the same object remained identical to those obtained with the unquantized YOLOv11-obb.pt model (Method 0). This consistency is unexpected.

**Comment by Y-T-G** [comment]
I am not sure how you are testing. Because from my testing there's a difference:

```python
In [56]: model = YOLO("yolo11n-obb.pt")

In [57]: result = model.val(data="dota8.yaml")                Ultralytics 8.3.0 🚀 Python-3.10.14 torch-2.4.1+cu121 CPU (AMD EPYC 7V12 64-Core Processor)
YOLO11n-obb summary (fused): 257 layers, 2,656,648 parameters, 0 gradients, 6.6 GFLOPs
val: Scanning /datasets/dota8/labels/val.cache... 4 images, 0
                 Class     Images  Instances      Box(P                          all          4          8      0.946          1      0.995      0.804
      baseball diamond          3          4      0.901          1      0.995       0.85
      basketball court          1          3      0.972          1      0.995      0.866                                         soccer ball field          1          1      0.965          1      0.995      0.697                                    ^[[5~Speed: 1.1ms preprocess, 78.7ms inference, 0.0ms loss, 1.4ms postprocess per image

In [58]: model = YOLO("yolo11n-obb.onnx")
In [60]: result = model.val(data="dota8.yaml", imgsz=1024)
Ultralytics 8.3.0 🚀 Python-3.10.14 torch-2.4.1+cu121 CPU (AMD EPYC 7V12 64-Core Processor)
Loading yolo11n-obb.onnx for ONNX Runtime inference...
Setting batch=1 input of shape (1, 3, 1024, 1024)
val: Scanning /datasets/dota8/labels/val.cache... 4 images, 0                  Class     Images  Instances      Box(P
                   all          4          8      0.979          1      0.995       0.82                                          baseball diamond          3          4      0.982          1      0.995      0.834
      basketball court          1          3      0.985          1      0.995      0.831
     soccer ball field          1          1       0.97          1      0.995      0.796
Speed: 4.1ms preprocess, 178.2ms inference, 0.0ms loss, 3.1ms postprocess per image
```

**Comment by Y-T-G** [comment]
@kuber-petlibro ONNX export doesn't support int8. FP16 export is supported but you also have to pass `device=0` along with `half=True`.

**Comment by Y-T-G** [comment]
For `half=True` with ONNX, you also need to pass a GPU device for it work. It should a warning for that.
